command/base.go

```diff
@@ -270,32 +269,20 @@ func (c *BaseCommand) validateMFA(reqID string, methodInfo MFAMethodInfo) int {
 
 	client, err := c.Client()
 	if err != nil {
-		c.UI.Error(err.Error())
-		return 2
+		return nil, err
 	}
 
 	secret, err := client.Sys().MFAValidate(reqID, mfaPayload)
 	if err != nil {
-		c.UI.Error(err.Error())
-		if secret != nil {
-			OutputSecret(c.UI, secret)
-		}
-		return 2
-	}
-	if secret == nil {
-		// Don't output anything unless using the "table" format
-		if Format(c.UI) == "table" {
-			c.UI.Info("Success! Data written to: sys/mfa/validate")
-		}
-		return 0
+		return secret, err
 	}
 
-	// Handle single field output
-	if c.flagField != "" {
-		return PrintRawField(c.UI, secret, c.flagField)
+	// Don't output anything unless using the "table" format
```

Comment:
Would you please point me to the output for the table format as noted in the comment? I could not find it.

Reply:
This just comes from here: https://github.com/hashicorp/vault/pull/17040/commits/6a98e9955765249b16d4a433036787aabe9c433f#diff-2176c69dbb39c3c2e412b713b90b79b52de3d280dff139fa58700ed976572c83L286. I believe that just comes from the `LoginCommand` `FlagSetOutputFormat`, right?

Reply:
So, this seems to me that this message should only be written if `MFAValidate` returns nil,nil. Meaning that there is no error, but then the secret is nil as well. So, printing this should at least check for nil, nil. But, then I am wondering why do we print success! Although validation seems to be successful, but there is no token returned! What do you think?

Reply:
I think in this case, you get `secret, nil`, which would be a success. (see: line 281) The caller `MFAValidate` should only return a nil secret in the error condition, I think.

Reply:
Seems like `MFAValidate` would never return nil,nil. 
So, I don't think this message is printed in the old code at all. And, now it is printed if error is nil. So, I am wondering if it would break some CLI clients who are parsing the output and encounter an additional string. Should we print this at all?

Reply:
Very good point! I'll drop this, thanks!